src/main/java/io/github/genomicdatainfrastructure/discovery/filters/infrastructure/beacon/BeaconFilterBuilder.java

```diff
@@ -0,0 +1,74 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package io.github.genomicdatainfrastructure.discovery.filters.infrastructure.beacon;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.github.genomicdatainfrastructure.discovery.datasets.infrastructure.beacon.auth.BeaconAuth;
+import io.github.genomicdatainfrastructure.discovery.filters.application.ports.FilterBuilder;
+import io.github.genomicdatainfrastructure.discovery.filters.infrastructure.beacon.strategies.FilterStrategyContext;
+import io.github.genomicdatainfrastructure.discovery.model.Filter;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.api.BeaconQueryApi;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconFilteringTermsResponse;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconFilteringTermsResponseContent;
+import io.quarkus.arc.lookup.LookupIfProperty;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static java.util.Optional.ofNullable;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+
+@ApplicationScoped
+@LookupIfProperty(name = "sources.beacon", stringValue = "true")
+public class BeaconFilterBuilder implements FilterBuilder {
+
+    private final BeaconQueryApi beaconQueryApi;
+    private final BeaconAuth beaconAuth;
+    private FilterStrategyContext context;
+
+    @Inject
+    public BeaconFilterBuilder(@RestClient BeaconQueryApi beaconQueryApi, BeaconAuth beaconAuth,
+            FilterStrategyContext context) {
+        this.beaconQueryApi = beaconQueryApi;
+        this.beaconAuth = beaconAuth;
+        this.context = context;
+    }
+
+    @Override
+    public List<Filter> build(String accessToken) {
+        var beaconAuthorization = beaconAuth.retrieveAuthorization(accessToken);
+        if (beaconAuthorization == null) {
+            return List.of();
+        }
+        var filteringTermsResponse = retrieveNonNullFilteringTermsResponse(beaconAuthorization);
+
+        return context.buildFilters(filteringTermsResponse);
```

Comment:
important: can you please review code?

We don't know how beacon filtering terms will look like in the future. For the sake of simplicity, we should not rely on this API anymore, so you can simply have a json with the expected filters in the format of our API directly, no need to parse it.